<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Sven Buerki - Boise State University" />

<meta name="date" content="2023-08-18" />

<title>EEB 603 – Reproducible Science</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Reproducible Science</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Timetable.html">Timetable</a>
</li>
<li>
  <a href="Chapters.html">Chapters</a>
</li>
<li>
  <a href="Resources.html">Resources</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">EEB 603 – Reproducible Science</h1>
<h3 class="subtitle">Syllabus</h3>
<h4 class="author">Sven Buerki - Boise State University</h4>
<h4 class="date">2023-08-18</h4>

</div>


<script type="text/javascript">
$(document).ready(function() {
  $('#header').parent().prepend('<div id=\"logo\"><img src=\"Images/boisestate-primarylogo-2color-rgb.png\" style=\"position:absolute; top:0; right:0; padding:10px; height:120px\"></div>');
  $('#header').css('margin-right', '120px')
});
</script>
<div style="text-align: right">
<a href="Data/EEB603_Syllabus_BUERKI_2023.pdf">Download pdf version</a>
</div>
<div style="text-align: right">
<a href="https://github.com/svenbuerki/EEB603_Reproducible_Science">Raw data on GitHub</a>
</div>
<div id="instructor" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Instructor</h1>
<ul>
<li>Name: Sven Buerki (he/his)</li>
<li>Office: Science building, office 114 (ground floor).</li>
<li>Email: <a href="mailto:svenbuerki@boisestate.edu" class="email">svenbuerki@boisestate.edu</a></li>
<li>Office hours: By appointment.</li>
</ul>
</div>
<div id="our-class-ethos" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Our class ethos</h1>
<blockquote>
<p>Everyone here is smart; distinguish yourself by being kind.</p>
<footer>
Kindness in Science is an inclusive approach that fosters diversity, respect, wellbeing &amp; openness leading to better science outcomes.
</footer>
</blockquote>
<p><img src="Images/Kindness.png" width="40%" style="display: block; margin: auto 0 auto auto;" /></p>
<p>Learn more about this initiative <a href="https://doi.org/10.1038/d41586-018-00482-y">here</a>, as well as a recent article delving into what kindness in science is and why it does matter <span class="citation">(<a href="#ref-Boulter2023" role="doc-biblioref">Boulter et al., 2023</a>)</span>.</p>
</div>
<div id="class-details" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Class details</h1>
<div id="teaching-mode-and-class-location" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Teaching mode and class location</h2>
<p>In-person in <a href="https://maps.boisestate.edu/?id=715#!m/89069">Science Bldg, Rm 149</a>.</p>
</div>
<div id="meeting-days-and-times" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Meeting days and times</h2>
<p>Tuesday &amp; Thursday from 9:00 to 10:15 AM.</p>
</div>
<div id="timetable" class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> Timetable</h2>
<p>The tentative timetable for this course is available <a href="Timetable.html">here</a>. The instructor wants to warn students that he might adjust the timetable to accommodate any needs. However, in case of changes, the instructor will make sure to contact enrolled students to keep them posted.</p>
</div>
<div id="health-and-safety-protocols" class="section level2" number="3.4">
<h2><span class="header-section-number">3.4</span> Health and safety protocols</h2>
<p><strong>Our priority is your safety!</strong></p>
<p>There are currently no specific health guidelines, but for more information on this topic please visit the <a href="https://www.boisestate.edu/publichealth/">BSU public health response</a> webpage.</p>
</div>
</div>
<div id="course-goal-description" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Course goal &amp; description</h1>
<p>The scientific community widely acknowledges that we are in the midst of a reproducibility crisis <span class="citation">(see e.g. <a href="#ref-Baker_2016" role="doc-biblioref">Baker, 2016</a>)</span>. This course starts by reviewing the evidence and causes supporting this crisis and aims at highlighting factors to boost reproducibility in science, especially in the fields of Ecology, Evolution &amp; Behavior. This course also aims at providing a platform to develop, practice and strengthen science communication skills <span class="citation">(see e.g. <a href="#ref-Soltis2023" role="doc-biblioref">Soltis et al., 2023</a> for some innovative ways to share your research)</span>.</p>
<p>The overarching aim of this course is therefore to provide students with the theoretical knowledge and bioinformatic tools necessary to improve transparency, reproducibility and efficiency in scientific research. Across the course of the course, students will be taught how to use open source software for their research such as <a href="https://cran.r-project.org">R</a>, <a href="https://www.rstudio.com">RStudio</a> and <a href="https://rmarkdown.rstudio.com">R Markdown</a> (incl. <a href="https://kbroman.org/knitr_knutshell/">knitr</a>). To further master these latter skills and improve communication and teaching skills, students will <em>i</em>) design bioinformatic tutorials and teach those to their peers and <em>ii</em>) design and implement individual projects aiming at developing a reproducible workflow specific to their research interest.</p>
<p>Overall, this course <strong>provides students with key knowledge to gather, store, share, prepare and analyze data as well as communicate results to the scientific community and various stakeholders</strong> (see Figure <a href="#fig:part2">4.1</a>).</p>
<div class="figure"><span style="display:block;" id="fig:part2"></span>
<img src="Images/Bioinformatic_workflow_PART2.png" alt="Overview of workflow studied in PART2." width="100%" />
<p class="caption">
Figure 4.1: Overview of workflow studied in PART2.
</p>
</div>
</div>
<div id="structure-of-the-course" class="section level1" number="5">
<h1><span class="header-section-number">5</span> <a name="Structure"></a>Structure of the course</h1>
<p>The course is subdivided into three parts:</p>
<ul>
<li><em>PART 1: The Big Picture</em></li>
<li><em>PART 2: Bioinformatics for Reproducible Science</em></li>
<li><em>PART 3: Apply a Reproducible Approach to your Data</em></li>
</ul>
<p>Part 1 provides students with key theoretical knowledge on reproducible science allowing them to successfully design and implement a reproducible approach tailored to Ecology, Evolution &amp; Behavior. This part will also cover material associated with open science and data management and how these practices intertwine with scientific publications.</p>
<p>Part 2 provides students with opportunities to further learn and apply the coding and bioinformatic tools required to implement a reproducible workflow (Figure <a href="#fig:part2">4.1</a>). Here, students (with the support of the instructor) will develop and teach a tutorial on a specific bioinformatic subject, which will be taught over two full classes. Tutorials will be written in RMarkdown and distributed to the class one week in advance. Depending on class size, this assignment will be conducted in group or individually.</p>
<p>Finally, part 3 was designed to provide students with an opportunity to develop individual reproducible workflows tailored to their research project (or to data presented in a publication if they don’t have a thesis subject yet). This assignment will be done by applying knowledge gained during the previous parts and working in collaboration with the instructor (in some cases, we might seek support from thesis advisers).</p>
<div id="content-of-the-course" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Content of the course</h2>
<p><strong>PART 1: The Big Picture</strong></p>
<ol style="list-style-type: decimal">
<li><strong>Introduction: The reproducibility crisis &amp; prospects to tackle it!</strong>
<ul>
<li>Chapter 1: Introduction to R, RStudio, Markdown (incl. referencing) &amp; User-defined functions.</li>
<li>Chapter 2: The reproducibility crisis.</li>
<li>Chapter 3: A roadmap to implement reproducible science in Ecology, Evolution &amp; Behavior.</li>
<li>Chapter 4: Open science and CARE principles.</li>
</ul></li>
<li><strong>Getting started: Overview of data workflow &amp; used software</strong>
<ul>
<li>Chapter 5: Data management, Reproducible code.</li>
<li>Chapter 6: Getting published &amp; Peer review.</li>
</ul></li>
</ol>
<p><a name="PART2"></a><strong>PART 2: Bioinformatics for Reproducible Science</strong></p>
<ol start="3" style="list-style-type: decimal">
<li><strong>Prepare your data</strong>
<ul>
<li>Chapter 7: Organize and import data with R.</li>
<li>Chapter 8: Prepare/tidy data for analyses in R.</li>
</ul></li>
<li><strong>Explore &amp; understand your data</strong>
<ul>
<li>Chapter 9: Statistical modelling and <em>knitr</em>.</li>
<li>Chapter 10: Visualize results with Tables.</li>
<li>Chapter 11: Visualize results with Figures (incl. handling phylogenetic trees).</li>
</ul></li>
<li><strong>Communicate/Disseminate your results</strong>
<ul>
<li>Chapter 12: Git &amp; GitHub: What are those and how can they help you with your code and communicating your research?</li>
</ul></li>
</ol>
<p><strong>PART 3: Apply a Reproducible Approach to your Data</strong></p>
<ol start="6" style="list-style-type: decimal">
<li><strong>Students develop and present reproducible workflows applied to their project</strong>
<ul>
<li>Produce individual reports showcasing reproducible workflows tailored to thesis projects.</li>
<li>Oral presentations of individual students’ projects.</li>
</ul></li>
</ol>
</div>
</div>
<div id="shared-google-drive" class="section level1" number="6">
<h1><span class="header-section-number">6</span> <a name="GoogleSite"></a>Shared Google Drive</h1>
<p>A shared Google drive is set up to facilitate data sharing and allow students uploading their <a href="#BioTut">bioinformatic tutorials</a>.</p>
<ul>
<li>URL: <a href="https://drive.google.com/drive/folders/1eZyjGQ478vaKiGD01hcryyNtdGp9t1J7?usp=sharing" class="uri">https://drive.google.com/drive/folders/1eZyjGQ478vaKiGD01hcryyNtdGp9t1J7?usp=sharing</a></li>
</ul>
</div>
<div id="publications-textbooks" class="section level1" number="7">
<h1><span class="header-section-number">7</span> <a name="Publis"></a>Publications &amp; Textbooks</h1>
<p>The reading material at the basis of this course is composed of a mixture of publications and chapters mostly from two textbooks <span class="citation">(<a href="#ref-Gandrud2015" role="doc-biblioref">Gandrud, 2015</a>; <a href="#ref-Wickham2017" role="doc-biblioref">Wickham and Grolemund, 2017</a>)</span>. We will also study the <em>“Guides to”</em> published by the <a href="https://www.britishecologicalsociety.org/publications/guides-to/">British Ecological Society</a>. Please find below the references used in each <a href="#Structure">chapter</a>. <strong>This list is not exhaustive and additional literature will be provided in class.</strong></p>
<table>
<colgroup>
<col width="23%" />
<col width="76%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Chapter</th>
<th>Reference(s)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><strong>Chap. 1</strong></td>
<td>Chapter 3 of <span class="citation">Gandrud (<a href="#ref-Gandrud2015" role="doc-biblioref">2015</a>)</span></td>
</tr>
<tr class="even">
<td align="left"><strong>Chap. 2</strong></td>
<td><span class="citation">Baker (<a href="#ref-Baker_2016" role="doc-biblioref">2016</a>)</span>; <span class="citation">Freedman et al. (<a href="#ref-Freedman_2015" role="doc-biblioref">2015</a>)</span>; <span class="citation">Munafo et al. (<a href="#ref-Munafo" role="doc-biblioref">2017</a>)</span>; <span class="citation">Peng and Hicks (<a href="#ref-Peng2021" role="doc-biblioref">2021</a>)</span>; <span class="citation">Sarewitz (<a href="#ref-Sarewitz_2016" role="doc-biblioref">2016</a>)</span></td>
</tr>
<tr class="odd">
<td align="left"><strong>Chap. 3</strong></td>
<td><span class="citation">Bone et al. (<a href="#ref-Bone2015" role="doc-biblioref">2015</a>)</span>; <span class="citation">Markowetz (<a href="#ref-Markowetz2015" role="doc-biblioref">2015</a>)</span>; <span class="citation">Smith et al. (<a href="#ref-Smith2016" role="doc-biblioref">2016</a>)</span></td>
</tr>
<tr class="even">
<td align="left"><strong>Chap. 4</strong></td>
<td><span class="citation">Carroll et al. (<a href="#ref-Carroll2021" role="doc-biblioref">2021</a>)</span>; <a href="https://creativecommons.org/">Creative Commons</a>; <span class="citation">Wagner et al. (<a href="#ref-Wagner2022" role="doc-biblioref">2022</a>)</span>; <span class="citation">Williams et al. (<a href="#ref-Williams2023" role="doc-biblioref">2023</a>)</span></td>
</tr>
<tr class="odd">
<td align="left"><strong>Chap. 5</strong></td>
<td><span class="citation">British Ecological Society (<a href="#ref-DataMan" role="doc-biblioref">2014a</a>)</span> &amp; Chapter 4 of <span class="citation">Gandrud (<a href="#ref-Gandrud2015" role="doc-biblioref">2015</a>)</span>; <span class="citation">British Ecological Society (<a href="#ref-RepCode" role="doc-biblioref">2014d</a>)</span> &amp; Chapter 2 of <span class="citation">Gandrud (<a href="#ref-Gandrud2015" role="doc-biblioref">2015</a>)</span>; <span class="citation">Trisovic et al. (<a href="#ref-Trisovic2022" role="doc-biblioref">2022</a>)</span></td>
</tr>
<tr class="even">
<td align="left"><strong>Chap. 6</strong></td>
<td><span class="citation">British Ecological Society (<a href="#ref-GetPub" role="doc-biblioref">2014b</a>)</span>; <span class="citation">British Ecological Society (<a href="#ref-PeerRev" role="doc-biblioref">2014c</a>)</span></td>
</tr>
<tr class="odd">
<td align="left"><strong>Chap. 7</strong></td>
<td>Chapter 6 of <span class="citation">Gandrud (<a href="#ref-Gandrud2015" role="doc-biblioref">2015</a>)</span></td>
</tr>
<tr class="even">
<td align="left"><strong>Chap. 8</strong></td>
<td>Chapter 7 of <span class="citation">Gandrud (<a href="#ref-Gandrud2015" role="doc-biblioref">2015</a>)</span> &amp; Chapters 9-10 of <span class="citation">Wickham and Grolemund (<a href="#ref-Wickham2017" role="doc-biblioref">2017</a>)</span></td>
</tr>
<tr class="odd">
<td align="left"><strong>Chap. 9</strong></td>
<td>Chapter 8 of <span class="citation">Gandrud (<a href="#ref-Gandrud2015" role="doc-biblioref">2015</a>)</span></td>
</tr>
<tr class="even">
<td align="left"><strong>Chap. 10</strong></td>
<td>Chapter 9 of <span class="citation">Gandrud (<a href="#ref-Gandrud2015" role="doc-biblioref">2015</a>)</span></td>
</tr>
<tr class="odd">
<td align="left"><strong>Chap. 11</strong></td>
<td>Chapter 10 of <span class="citation">Gandrud (<a href="#ref-Gandrud2015" role="doc-biblioref">2015</a>)</span>, Chapters 1 and 22 of <span class="citation">Wickham and Grolemund (<a href="#ref-Wickham2017" role="doc-biblioref">2017</a>)</span> &amp; <span class="citation">Guangchuang et al. (<a href="#ref-Yu2017" role="doc-biblioref">2017</a>)</span></td>
</tr>
<tr class="even">
<td align="left"><strong>Chap. 12</strong></td>
<td>Chapter 5 of <span class="citation">Gandrud (<a href="#ref-Gandrud2015" role="doc-biblioref">2015</a>)</span></td>
</tr>
</tbody>
</table>
</div>
<div id="examples-of-reproducible-science" class="section level1" number="8">
<h1><span class="header-section-number">8</span> Examples of reproducible science</h1>
<p>To further exemplify how a reproducible approach can be implemented in your research, the instructor provides here examples of publications and software produced by students that have attended this course (alphabetically sorted):</p>
<ul>
<li><span class="citation">Cole et al. (<a href="#ref-Cole2022" role="doc-biblioref">2022</a>)</span> – <strong>EcoCountHelper: an R package and analytical pipeline for the analysis of ecological count data using GLMMs, and a case study of bats in Grand Teton National Park.</strong></li>
<li><span class="citation">Ellestad et al. (<a href="#ref-Ellestad2022" role="doc-biblioref">2022</a>)</span> – <strong>Genomic Insights into Cultivated Mexican Vanilla planifolia Reveal High Levels of Heterozygosity Stemming from Hybridization.</strong></li>
<li><span class="citation">Wojahn et al. (<a href="#ref-Wojahn2021" role="doc-biblioref">2021</a>)</span> – <strong>G2PMineR: A Genome to Phenome Literature Review Approach.</strong></li>
</ul>
</div>
<div id="the-computing-tools-of-reproducible-science" class="section level1" number="9">
<h1><span class="header-section-number">9</span> <a name="software"></a>The computing tools of reproducible science</h1>
<p>Research is often presented in the form of slideshows, articles or books. These presentation documents announce a project’s findings, but they are not the research, they are the advertisement part of the research project!</p>
<blockquote>
<p><em>The research is the full software environment, code, and data that produced the results <span class="citation">(<a href="#ref-Donoho2010" role="doc-biblioref">Donoho, 2010</a>)</span>.</em></p>
</blockquote>
<p><strong>When we separate the research from its advertisement, we are making it difficult for others to verify the findings by reproducing them.</strong></p>
<p>This course will give you the tools to dynamically combine your research with the presentation of your findings. The first tool will be a workflow for reproducible research weaving the principles of reproducibility throughout your entire research project, from data gathering to the statistical analysis, and the presentation of results. To reach this goal, you will learn how to use a number of computer tools that make this workflow possible.</p>
<div id="the-bioinformatic-tools" class="section level2" number="9.1">
<h2><span class="header-section-number">9.1</span> The bioinformatic tools</h2>
<p>The main bioinformatic tools covered in this course are:</p>
<ul>
<li>The <strong>R</strong> statistical language that will allow you to gather data and analyze it.</li>
<li>The <strong>LaTeX</strong> and <strong>Markdown</strong> markup languages that you can use to create documents (slideshows, articles, books, webpages) for presenting your findings.</li>
<li>The <em>knitr</em> and <em>rmarkdown</em> <strong>packages</strong> for R and other tools, including <strong>command-line shell programs</strong> like GNU Make and Git version control, for dynamically tiding your data gathering, analysis, and presentation documents together so that they can be easily reproduced.</li>
<li><strong>RStudio</strong>, a program that brings all of these tools together in one place.</li>
</ul>
</div>
<div id="installing-the-main-software" class="section level2" number="9.2">
<h2><span class="header-section-number">9.2</span> Installing the main software</h2>
<p>As shown above, <strong>R</strong> and <strong>RStudio</strong> are at the core of this course and will have to be installed on your computers. This can be easily done by downloading the software from the following websites:</p>
<ul>
<li><strong>R</strong>: <a href="https://www.r-project.org" class="uri">https://www.r-project.org</a></li>
<li><strong>RStudio</strong>: <a href="https://www.rstudio.com/products/rstudio/download/" class="uri">https://www.rstudio.com/products/rstudio/download/</a></li>
</ul>
<p>The download webpages for these software have comprehensive information on how to install them, so please refer to those pages for more information.</p>
</div>
<div id="installing-markup-languages" class="section level2" number="9.3">
<h2><span class="header-section-number">9.3</span> Installing markup languages</h2>
<p>If you are planning to create LaTeX documents, you will need to install a Tex distribution. Please refer to this website for more details: <a href="https://www.latex-project.org/get/" class="uri">https://www.latex-project.org/get/</a></p>
<p>If you want to create Markdown documents you can separately install the <em>rmarkdown</em> package in R (see <a href="#Rpack">below</a> for more details).</p>
</div>
<div id="installing-r-packages" class="section level2" number="9.4">
<h2><span class="header-section-number">9.4</span> <a name="Rpack"></a>Installing R packages</h2>
<p>We will be using a number of R packages especially designed to support reproducible research. Many of those packages are not included in the default R installation and will need to be installed separately.
To install key packages used in class, copy the following code and paste it into your R console:</p>
<pre class="r"><code>install.packages(c(&quot;brew&quot;, &quot;countrycode&quot;, &quot;devtools&quot;, &quot;dplyr&quot;, &quot;ggplot2&quot;, &quot;googleVis&quot;,
    &quot;knitr&quot;, &quot;rmarkdown&quot;, &quot;tidyr&quot;, &quot;xtable&quot;))</code></pre>
<p>Once you enter this code, you may be asked to select a CRAN “mirror” to download the packages from. Simply select the mirror closest to you.</p>
<p>Finally, it is highly likely that we will have to install additional packages. In this case, you can simply install it by using the same R function <code>install.packages()</code> or by using RStudio as follows: Select “Tools” -&gt; “Install Packages …” and then type the name of the package in the window (make sure to tick the “Install dependencies” box).</p>
</div>
</div>
<div id="rstudio-cheat-sheets" class="section level1" number="10">
<h1><span class="header-section-number">10</span> <a name="Cheat"></a>RStudio Cheat Sheets</h1>
<p>RStudio provides a suite of cheat sheets that can be accessed by going to the “Help” menu and selecting “Cheatsheets”.</p>
<p>Five cheat sheets are especially relevant to chapters taught in this course:</p>
<ul>
<li><em>RStudio IDE : Cheat sheet</em></li>
<li><em>Data Manipulation with dplyr, tidyr</em></li>
<li><em>Data Visualization with ggplot2</em></li>
<li><em>R Markdown Cheat Sheet</em></li>
<li><em>R Markdown Reference Guide</em></li>
</ul>
<p>These documents together with the material presented in <a href="#Publis">publications &amp; textbooks</a> will provide the basis to design your <a href="#BioTut">bioinformatic tutorials</a>.</p>
</div>
<div id="r-tutorials" class="section level1" number="11">
<h1><span class="header-section-number">11</span> R tutorials</h1>
<p>Please find below two documents providing a comprehensive introduction to R:</p>
<ul>
<li>R for beginners (a tutorial by Emmanuel Paradis): <a href="https://cran.r-project.org/doc/contrib/Paradis-rdebuts_en.pdf" class="uri">https://cran.r-project.org/doc/contrib/Paradis-rdebuts_en.pdf</a></li>
<li>An introduction to R: <a href="https://cran.r-project.org/doc/manuals/r-release/R-intro.pdf" class="uri">https://cran.r-project.org/doc/manuals/r-release/R-intro.pdf</a></li>
</ul>
</div>
<div id="assessments" class="section level1" number="12">
<h1><span class="header-section-number">12</span> Assessments</h1>
<p>There will not be any classical exams in this course, but we will rather focus on developing theoretical and bioinformatic skills and applying those to your research. In this context, each student will be asked to produce a bioinformatic tutorial and teach it to their peers (see <a href="#PART2">PART 2</a>). Each student will also be tasked to produce a report (tailored to their thesis project or a publication) and present their results and conclusions in class.</p>
<div id="tests-conducted-during-the-course-grading" class="section level2" number="12.1">
<h2><span class="header-section-number">12.1</span> Tests conducted during the course &amp; grading</h2>
<p>Students will be graded based on the following four tasks:</p>
<ul>
<li>Produce a bioinformatic tutorial focusing on a chapter from <a href="#PART2">PART 2</a>. Depending on enrollment students may be working in pairs (150 points).</li>
<li>Teach a bioinformatic lab (spread across 2 sessions; 100 points).</li>
<li>Produce an individual report on thesis project/publication (200 points).</li>
<li>One oral presentation on thesis project/publication (100 points).</li>
</ul>
<p>Exams are summing to a total of <strong>550 points</strong> and Table <a href="#tab:mygrade">12.1</a> exhibits the grading scale applied in this course.</p>
<table class="table table-striped table-hover table-condensed" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:mygrade">Table 12.1: </span>Grading scale applied in this course.
</caption>
<thead>
<tr>
<th style="text-align:left;">
Percentage
</th>
<th style="text-align:left;">
Grade
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
100-98
</td>
<td style="text-align:left;">
A+
</td>
</tr>
<tr>
<td style="text-align:left;">
97.9-93
</td>
<td style="text-align:left;">
A
</td>
</tr>
<tr>
<td style="text-align:left;">
92.9-90
</td>
<td style="text-align:left;">
A-
</td>
</tr>
<tr>
<td style="text-align:left;">
89.9-88
</td>
<td style="text-align:left;">
B+
</td>
</tr>
<tr>
<td style="text-align:left;">
87.9-83
</td>
<td style="text-align:left;">
B
</td>
</tr>
<tr>
<td style="text-align:left;">
82.9-80
</td>
<td style="text-align:left;">
B-
</td>
</tr>
<tr>
<td style="text-align:left;">
79.9-78
</td>
<td style="text-align:left;">
C+
</td>
</tr>
<tr>
<td style="text-align:left;">
77.9-73
</td>
<td style="text-align:left;">
C
</td>
</tr>
<tr>
<td style="text-align:left;">
72.9-70
</td>
<td style="text-align:left;">
C-
</td>
</tr>
<tr>
<td style="text-align:left;">
69.9-68
</td>
<td style="text-align:left;">
D+
</td>
</tr>
<tr>
<td style="text-align:left;">
67.9-60
</td>
<td style="text-align:left;">
D
</td>
</tr>
<tr>
<td style="text-align:left;">
59.9-0
</td>
<td style="text-align:left;">
F
</td>
</tr>
</tbody>
</table>
</div>
<div id="bioinformatic-tutorial-150-points" class="section level2" number="12.2">
<h2><span class="header-section-number">12.2</span> <a name="BioTut"></a>Bioinformatic tutorial (150 points)</h2>
<p>During week 1, students will be assigned a chapter of <a href="#PART2">PART 2</a> to study and produce a bioinformatic tutorial. Based on enrollment, students might work individually or in pairs (please see below for more information).</p>
<p>Tutorials will have to be written in the <em>knitr</em>/<em>rmarkdown</em> language as implemented in RStudio. Tutorials should be focused on developing a suite of exercises aiming at gaining key bioinformatic skills specific to each chapter (see <a href="#PART2">PART 2</a>). Students will be welcomed to use material presented in <span class="citation">Gandrud (<a href="#ref-Gandrud2015" role="doc-biblioref">2015</a>)</span> and <span class="citation">Wickham and Grolemund (<a href="#ref-Wickham2017" role="doc-biblioref">2017</a>)</span> to develop their tutorials, but they can also use other sources as long as they are properly cited in their documents. See <a href="#Publis">Publications &amp; Textbooks</a> and <a href="#Cheat">RStudio Cheat Sheets</a> sections for more details.</p>
<p>Students should design their tutorials to be completed within 2 laboratory sessions (see <a href="#TeachTut">below</a>). <strong>Tutorials should be submitted to the instructor 1 week in advance for correction and to be uploaded onto the shared Google drive.</strong></p>
<div id="what-do-you-need-to-keep-in-mind" class="section level3" number="12.2.1">
<h3><span class="header-section-number">12.2.1</span> What do you need to keep in mind?</h3>
<p>While designing your tutorials, please think about the following points:</p>
<ol style="list-style-type: decimal">
<li>What are your major objectives and how do they fit within the scientific process and data life-cycle? Please also focus on the theoretical background.</li>
<li>What are the specific learning outcomes?</li>
<li>What data are used to teach this tutorial?</li>
<li>What are the bioinformatic dependencies and how to install them?</li>
<li>When you teach each concept, make sure to clearly explain the rational, steps to get there (pseudo-code) and focus on the input/output and data class/format used for each function.</li>
</ol>
</div>
<div id="content-of-the-tutorial" class="section level3" number="12.2.2">
<h3><span class="header-section-number">12.2.2</span> Content of the tutorial</h3>
<p>Based on the information provided above, your tutorial should include:</p>
<ul>
<li>A short introduction highlighting the theory and aims of the tutorial.</li>
<li>A section on R package requirements with instructions on how to install those packages and their dependencies.</li>
<li>A section introducing the data (dataset) used to support these exercises (and how to download those).</li>
<li>A references section and links to manuals of R packages.</li>
<li>Commented R code necessary to guide users through the exercises as well as some knowledge on expected outputs.</li>
</ul>
</div>
<div id="google-sheet-with-groups-for-bioinformatic-tutorials" class="section level3" number="12.2.3">
<h3><span class="header-section-number">12.2.3</span> Google sheet with groups for bioinformatic tutorials</h3>
<p>The instructor is asking students to sign up for designing and teaching a bioinformatic tutorial by accessing the following Google sheet (the number of students per chapter is provided, please respect this guideline):</p>
<ul>
<li><a href="https://docs.google.com/spreadsheets/d/1gUwbqJ4gbhhSY1DfsZ7HsoDLzOKB-57xAAC1_IJ3qmk/edit?usp=sharing" class="uri">https://docs.google.com/spreadsheets/d/1gUwbqJ4gbhhSY1DfsZ7HsoDLzOKB-57xAAC1_IJ3qmk/edit?usp=sharing</a></li>
</ul>
</div>
</div>
<div id="teaching-your-bioinformatic-tutorial-100-points" class="section level2" number="12.3">
<h2><span class="header-section-number">12.3</span> <a name="TeachTut"></a>Teaching your bioinformatic tutorial (100 points)</h2>
<p>Students are expected to prepare a 10 minutes presentation providing general guidelines to complete the tutorial. Presentations will be uploaded onto the shared Google drive and made accessible to all students. Students are expected to support their peers in completing tutorials by answering questions. The instructor will also be answering questions, but students are leading the teaching of the bioinformatic laboratories.</p>
<p>Students will be graded according to their abilities to teach their tutorials and answer questions. The instructor might also use student’s feedback to grade this test.</p>
</div>
<div id="individual-report-on-thesis-projectpublication-200-points" class="section level2" number="12.4">
<h2><span class="header-section-number">12.4</span> Individual report on thesis project/publication (200 points)</h2>
<p>Students will work alongside the instructor to develop a reproducible workflow specific to their thesis project. In cases where students do not yet have a clear idea on their thesis project, they will work with instructor to identify a publication that can serve as basis for their individual project.</p>
<p>Reports will be written using the <em>knitr</em>/<em>rmarkdown</em> markup language as implemented in RStudio. The instructor expects students to provide a list of references supporting their reports. References will have to be cited in the text: it is not enough to through a bunch of references at the end of the report. This exercise aims at supporting methodological decisions taken in the report and increasing transparency.</p>
<div id="students-gather-prior-key-elements-on-their-project" class="section level3" number="12.4.1">
<h3><span class="header-section-number">12.4.1</span> Students gather prior key elements on their project</h3>
<ul>
<li>What kind of data are already published/available (and which of those are relevant to your topic)?</li>
<li>Where are those published data deposited?</li>
<li>Can you reproduce the analyses based on published data?</li>
<li>What types of data are or will be produced during your thesis project?</li>
<li>What are the specificity of your data (in term of storage, sharing, etc.)?</li>
<li>What are the publication standards in your field <span class="citation">(see e.g. <a href="#ref-Donoho2010" role="doc-biblioref">Donoho, 2010</a>; <a href="#ref-Smith2016" role="doc-biblioref">Smith et al., 2016</a>)</span>?</li>
<li>etc…</li>
</ul>
</div>
<div id="students-develop-the-following-core-processes-for-their-project" class="section level3" number="12.4.2">
<h3><span class="header-section-number">12.4.2</span> Students develop the following core processes for their project</h3>
<ul>
<li>A data management workflow specific to your research, which will cover the following stages of the data life cycle <span class="citation">(see e.g. <a href="#ref-DataMan" role="doc-biblioref">British Ecological Society, 2014a</a>)</span>:
<ul>
<li>Create</li>
<li>Process</li>
<li>Document</li>
<li>Preserve</li>
<li>Share</li>
<li>Reuse</li>
</ul></li>
<li>A reproducible code to perform the following tasks to your data (see Fig. <span class="math inline">\(\ref{fig:part2}\)</span>):
<ul>
<li>Import</li>
<li>Tidy/clean</li>
<li>Transform</li>
<li>Visualize</li>
<li>Model</li>
<li>Communicate</li>
</ul></li>
</ul>
</div>
</div>
<div id="oral-presentation-on-thesis-projectpublication-100-points" class="section level2" number="12.5">
<h2><span class="header-section-number">12.5</span> Oral presentation on thesis project/publication (100 points)</h2>
<p>Each student will have to present their report during final week. The presentation should follow the same structure as the report and not exceed 15 minutes. There will be 5 minutes at the end of the presentation allocated for questions.</p>
</div>
</div>
<div id="late-work-policy" class="section level1" number="13">
<h1><span class="header-section-number">13</span> Late work policy</h1>
<p>The instructor expects students to deliver their assignments on time and set enough time aside to work on their projects (see above for more details). However, if you have any issues preventing completion of your work on time, please contact the instructor as soon as possible to find common solutions.</p>
</div>
<div id="engagement-expectations" class="section level1" number="14">
<h1><span class="header-section-number">14</span> Engagement expectations</h1>
<p>The instructor is expecting students to attend classes (please join on time and for the full duration of the class) and actively engage by asking questions and giving feedback on teaching material and course content. This course was designed to help students implementing a reproducible approach to their research projects. If you are judging that additional content should be covered in class, please contact the instructor. The instructor will do his very best to obtain information or seek support from colleagues to cover the requested material. In the case that you have any issues attending class, please contact the instructor by email (<a href="mailto:svenbuerki@boisestate.edu" class="email">svenbuerki@boisestate.edu</a>) asap and see below for more details.</p>
</div>
<div id="what-you-can-expect-of-me" class="section level1" number="15">
<h1><span class="header-section-number">15</span> What you can expect of me</h1>
<p>The instructor will be prepared for class, on time and not leave early. He will also be respectful of you and your opinions. Overall, the instructor wants to foster a kind and respectful class environment where all students can express themselves and share their opinions. This means that meaningful and constructive dialogue is encouraged in this class and it requires a degree of mutual respect, willingness to listen, and tolerance of opposing points of view. Respect for individual differences and alternative viewpoints will be maintained at all times in this class. One’s words and use of language should be temperate and within acceptable bounds of civility and decency. Finally, the instructor will reply to emails and grade tests as soon as possible (and provide positive criticism) to allow students mastering the material presented in class.</p>
</div>
<div id="this-course-was-designed-with-you-in-mind" class="section level1" number="16">
<h1><span class="header-section-number">16</span> This course was designed with you in mind</h1>
<p>The instructor has developed this course to provide a welcoming environment and effective, equitable learning experience for all students. If you encounter barriers in this course, please bring them to my attention so that I may work to address them.</p>
<div id="this-classs-community-is-inclusive" class="section level2" number="16.1">
<h2><span class="header-section-number">16.1</span> This class’s community is inclusive</h2>
<p>Students in this class represent a rich variety of backgrounds and perspectives. The <a href="https://www.boisestate.edu/biology/">Biological Sciences department</a> is committed to providing an atmosphere for learning that respects diversity and creates inclusive environments in our courses. While working together to build this community, we ask all members to:</p>
<ul>
<li>Share their unique experiences, values, and beliefs, if comfortable doing so.</li>
<li>Listen deeply to one another.</li>
<li>Honor the uniqueness of their peers.</li>
<li>Appreciate the opportunity we have to learn from each other in this community.</li>
<li>Use this opportunity together to discuss ways in which we can create an inclusive environment in this course and across the campus community.</li>
<li>Recognize opportunities to invite a community member to exhibit more inclusive, equitable speech or behavior—and then also invite them into further conversation. We also expect community members to respond with gratitude and to take a moment of reflection when they receive such an invitation, rather than react immediately from defensiveness.</li>
<li>Keep confidential any discussions that the community has of a personal (or professional) nature, unless the speaker has given explicit permission to share what they have said.</li>
<li>Respect the right of students to be addressed and referred to by the names and pronouns that correspond to their gender identities, including the use of non-binary pronouns.</li>
</ul>
</div>
<div id="we-use-each-others-preferred-names-and-pronouns" class="section level2" number="16.2">
<h2><span class="header-section-number">16.2</span> We use each other’s preferred names and pronouns</h2>
<p>Please let the instructor know of your preferred or adopted name and gender pronoun(s), and he will make those changes to his own records and address you that way in all cases.</p>
<p>To change to a preferred name so that it displays on all BSU sites, including Canvas and the course roster, contact the Registrar’s Office at (208) 426-4249. Note that only a legal name change can alter your name on BSU official and legal documents (e.g., your transcript).</p>
</div>
<div id="this-course-is-accessible-to-students-with-disabilities" class="section level2" number="16.3">
<h2><span class="header-section-number">16.3</span> This course is accessible to students with disabilities</h2>
<p>The instructor recognizes that navigating your education and life can often be more difficult if you have disabilities. I want you to achieve at your highest capacity in this class. If you have a disability, the instructor needs to know if you encounter inequitable opportunities in this course related to:
- Accessing and understanding course materials.
- Engaging with course materials and other students in the course.
- Demonstrating your skills and knowledge on assignments and exams.</p>
<p>If you have a documented disability, you may be eligible for accommodations in all of your courses. To learn more, make an appointment with the university’s <a href="https://www.boisestate.edu/eac/">Educational Access Center</a>.</p>
</div>
<div id="for-students-responsible-for-children" class="section level2" number="16.4">
<h2><span class="header-section-number">16.4</span> For students responsible for children</h2>
<p>The instructor recognizes the unique challenges that can arise for students who are also parents or guardians of children. If you have any specific needs related to this topic, please contact the instructor asap.</p>
</div>
</div>
<div id="academic-integrity" class="section level1" number="17">
<h1><span class="header-section-number">17</span> Academic integrity</h1>
<p>To create a welcoming, engaging, and effective learning environment, the instructor expects all of us to exhibit behavior that reflects Boise State’s <a href="https://www.boisestate.edu/president/values/statement-of-shared-values/">Statement of Shared Values</a>. The Shared Values emphasize academic excellence, caring, citizenship, fairness, respect, responsibility, and trustworthiness. In keeping with these values, the instructor expects students in this course to uphold the standards outlined in the Boise State University <a href="https://www.boisestate.edu/policy/student-affairs/code-of-conduct/">Student Code of Conduct</a>.</p>
</div>
<div id="student-well-being" class="section level1" number="18">
<h1><span class="header-section-number">18</span> Student well-being</h1>
<p>If you are struggling for any reason (e.g., COVID, relationship, family, or life’s stresses) and believe these may impact your performance in the course, the instructor is encouraging you to contact the Dean of Students at (208) 426-1527 or email <a href="mailto:deanofstudents@boisestate.edu" class="email">deanofstudents@boisestate.edu</a> for support. Additionally, if you are comfortable doing so, please reach out to me and I will provide any resources or accommodations that I can. If you notice a significant change in your mood, sleep, feelings of hopelessness or a lack of self worth, consider connecting immediately with Counseling Services (1529 Belmont Street, Norco Building) at (208) 426-1459 or email <a href="mailto:healthservices@boisestate.edu" class="email">healthservices@boisestate.edu</a>.</p>
<div id="what-do-you-need" class="section level2" number="18.1">
<h2><span class="header-section-number">18.1</span> What do you need?</h2>
<p>The university has many resources designed to support you as a learner and human being. Among these are:</p>
<ul>
<li><a href="https://library.boisestate.edu/">Albertsons Library</a> provides a treasure trove of physical and electronic resources.
<ul>
<li>As you enter the library, straight ahead you’ll find the Reference Desk, where librarians can help you find the information and resources you need.</li>
<li>The Circulation Desk lets students borrow various technologies.</li>
<li>The MakerLab on the second floor offers tools for student use, and there are friendly staff in the MakerLab to help you learn how.</li>
</ul></li>
<li>The <a href="https://www.boisestate.edu/writingcenter/">Writing Center</a> offers individual consultations tailored to your needs, including making sense of writing assignment instructions, brainstorming, crafting a thesis, organizing an essay, revisions, citations, and more.</li>
<li><a href="https://www.boisestate.edu/healthservices/counseling">Counseling Services</a> helps you tap into your strengths and find resources to deal more effectively with concerns that impact your pursuit of personal and academic goals. It emphasizes prevention and early detection and provides a broad spectrum of short-term counseling, consultative, evaluative, teaching, and training functions. Counseling staff consists of licensed counselors, psychologists, and closely supervised trainees/post-graduate interns.</li>
<li>The <a href="https://www.boisestate.edu/genderequity/">Gender Equity Center</a> provides specialized, brief, no-cost, confidential support for campus members affected by:
<ul>
<li>Identity-related harassment.</li>
<li>Sexual harassment, sexual assault, unhealthy relationships, and stalking.</li>
<li>Life experiences related to gender, sexual orientation, and allyship.</li>
<li>Life transitions and crisis.</li>
<li>Personal, financial, or academic roadblocks.</li>
</ul></li>
<li><a href="https://www.boisestate.edu/mss/">Multicultural Student Services</a> works with all students interested in equity and social justice, students from different cultures, countries and ethnicities. The staff understands cultural differences and perspectives, and knows what it’s like living in a dominant culture. The training and the programs Multicultural Student Services provides can help you navigate through political correctness and into respectful dialogue. This center also provides:
<ul>
<li>Support and guidance for planning cultural festivals and events.</li>
<li>In-center tutoring and studying lounge.</li>
<li>Volunteer opportunities.</li>
<li>Mentoring and Training.</li>
<li>Programs and events.</li>
</ul></li>
<li>Food assistance: If you are hungry and cannot afford to purchase food, the campus has some resources to help you. You can visit the <a href="https://www.boisestate.edu/deanofstudents/student-support/food-help/food-pantry/">campus food pantry</a> or <a href="https://www.boisestate.edu/deanofstudents/student-support/food-help/meal-assistance/">get free meals in the campus dining hall</a>.</li>
</ul>
</div>
</div>
<div id="references" class="section level1" number="19">
<h1><span class="header-section-number">19</span> References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Baker_2016" class="csl-entry">
<span class="smallcaps">Baker, M.</span> 2016. 1,500 scientists lift the lid on reproducibility. <em>Nature</em> 533: 452–454. Available at: <a href="https://doi.org/10.1038/533452a">https://doi.org/10.1038/533452a</a>.
</div>
<div id="ref-Bone2015" class="csl-entry">
<span class="smallcaps">Bone, R.E.</span>, <span class="smallcaps">J.A.C. Smith</span>, <span class="smallcaps">N. Arrigo</span>, and <span class="smallcaps">S. Buerki</span>. 2015. A macro-ecological perspective on crassulacean acid metabolism (CAM) photosynthesis evolution in afro-madagascan drylands: Eulophiinae orchids as a case study. <em>New Phytologist</em> 208: 469–481. Available at: <a href="http://dx.doi.org/10.1111/nph.13572">http://dx.doi.org/10.1111/nph.13572</a>.
</div>
<div id="ref-Boulter2023" class="csl-entry">
<span class="smallcaps">Boulter, J.</span>, <span class="smallcaps">M.L. Orozco Morales</span>, <span class="smallcaps">N. Principe</span>, and <span class="smallcaps">C.M. Tilsed</span>. 2023. What is kindness in science and why does it matter? <em>Immunology &amp; Cell Biology</em> 101: 97–103. Available at: <a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/imcb.12580">https://onlinelibrary.wiley.com/doi/abs/10.1111/imcb.12580</a>.
</div>
<div id="ref-DataMan" class="csl-entry">
<span class="smallcaps">British Ecological Society</span> ed. 2014a. A guide to data management in ecology and evolution. British Ecological Society.
</div>
<div id="ref-GetPub" class="csl-entry">
<span class="smallcaps">British Ecological Society</span> ed. 2014b. A guide to getting published in ecology and evolution. British Ecological Society.
</div>
<div id="ref-PeerRev" class="csl-entry">
<span class="smallcaps">British Ecological Society</span> ed. 2014c. A guide to peer review in ecology and evolution. British Ecological Society.
</div>
<div id="ref-RepCode" class="csl-entry">
<span class="smallcaps">British Ecological Society</span> ed. 2014d. A guide to reproducible code in ecology and evolution. British Ecological Society.
</div>
<div id="ref-Carroll2021" class="csl-entry">
<span class="smallcaps">Carroll, S.R.</span>, <span class="smallcaps">E. Herczog</span>, <span class="smallcaps">M. Hudson</span>, <span class="smallcaps">K. Russell</span>, and <span class="smallcaps">S. Stall</span>. 2021. Operationalizing the CARE and FAIR principles for indigenous data futures. <em>Scientific Data</em> 8: 108. Available at: <a href="https://doi.org/10.1038/s41597-021-00892-0">https://doi.org/10.1038/s41597-021-00892-0</a>.
</div>
<div id="ref-Cole2022" class="csl-entry">
<span class="smallcaps">Cole, H.J.</span>, <span class="smallcaps">D.G. Gomes</span>, and <span class="smallcaps">J.R. Barber</span>. 2022. <span>EcoCountHelper</span>: An r package and analytical pipeline for the analysis of ecological count data using <span>GLMMs</span>, and a case study of bats in grant teton national park. <em><span>PeerJ</span></em> 10: e14509. Available at: <a href="https://doi.org/10.7717/peerj.14509">https://doi.org/10.7717/peerj.14509</a>.
</div>
<div id="ref-Donoho2010" class="csl-entry">
<span class="smallcaps">Donoho, D.L.</span> 2010. An invitation to reproducible computational research. <em>Biostatistics</em> 11: 385–388. Available at: <a href="http://dx.doi.org/10.1093/biostatistics/kxq028">http://dx.doi.org/10.1093/biostatistics/kxq028</a>.
</div>
<div id="ref-Ellestad2022" class="csl-entry">
<span class="smallcaps">Ellestad, P.</span>, <span class="smallcaps">M.A. Pérez-Farrera</span>, and <span class="smallcaps">S. Buerki</span>. 2022. Genomic insights into cultivated mexican vanilla planifolia reveal high levels of heterozygosity stemming from hybridization. <em>Plants</em> 11: 2090. Available at: <a href="https://doi.org/10.3390/plants11162090">https://doi.org/10.3390/plants11162090</a>.
</div>
<div id="ref-Freedman_2015" class="csl-entry">
<span class="smallcaps">Freedman, L.P.</span>, <span class="smallcaps">I.M. Cockburn</span>, and <span class="smallcaps">T.S. Simcoe</span>. 2015. The economics of reproducibility in preclinical research. <em><span>PLOS</span> Biology</em> 13: e1002165. Available at: <a href="https://doi.org/10.1371/journal.pbio.1002165">https://doi.org/10.1371/journal.pbio.1002165</a>.
</div>
<div id="ref-Gandrud2015" class="csl-entry">
<span class="smallcaps">Gandrud, C.</span> 2015. <span class="nocase">Reproducible Research with R and RStudio</span>. C. Gandrud [ed.],. CRC Press.
</div>
<div id="ref-Yu2017" class="csl-entry">
<span class="smallcaps">Guangchuang, Y.</span>, <span class="smallcaps">S.D. K.</span>, <span class="smallcaps">Z. Huachen</span>, <span class="smallcaps">G. Yi</span>, and <span class="smallcaps">L.T.T. Yuk</span>. 2017. Ggtree: An r package for visualization and annotation of phylogenetic trees with their covariates and other associated data. <em>Methods in Ecology and Evolution</em> 8: 28–36. Available at: <a href="https://besjournals.onlinelibrary.wiley.com/doi/abs/10.1111/2041-210X.12628">https://besjournals.onlinelibrary.wiley.com/doi/abs/10.1111/2041-210X.12628</a>.
</div>
<div id="ref-Markowetz2015" class="csl-entry">
<span class="smallcaps">Markowetz, F.</span> 2015. Five selfish reasons to work reproducibly. <em>Genome Biology</em> 16: 274. Available at: <a href="https://doi.org/10.1186/s13059-015-0850-7">https://doi.org/10.1186/s13059-015-0850-7</a>.
</div>
<div id="ref-Munafo" class="csl-entry">
<span class="smallcaps">Munafo, M.R.</span>, <span class="smallcaps">B.A. Nosek</span>, <span class="smallcaps">D.V.M. Bishop</span>, <span class="smallcaps">K.S. Button</span>, <span class="smallcaps">C.D. Chambers</span>, <span class="smallcaps">N.P. du Sert</span>, <span class="smallcaps">U. Simonsohn</span>, et al. 2017. <a href="https://doi.org/{10.1038/s41562-016-0021}"><span class="nocase">A manifesto for reproducible science</span></a>. <em><span>Nature human behaviour</span></em> <span>1</span>: 0021.
</div>
<div id="ref-Peng2021" class="csl-entry">
<span class="smallcaps">Peng, R.D.</span>, and <span class="smallcaps">S.C. Hicks</span>. 2021. Reproducible research: A retrospective. <em>Annual Review of Public Health</em> 42: 79–93. Available at:<a href=" 
        https://doi.org/10.1146/annurev-publhealth-012420-105110
    
">
https://doi.org/10.1146/annurev-publhealth-012420-105110
</a>.
</div>
<div id="ref-Sarewitz_2016" class="csl-entry">
<span class="smallcaps">Sarewitz, D.</span> 2016. The pressure to publish pushes down quality. <em>Nature</em> 533: 147–147. Available at: <a href="https://doi.org/10.1038/533147a">https://doi.org/10.1038/533147a</a>.
</div>
<div id="ref-Smith2016" class="csl-entry">
<span class="smallcaps">Smith, J.F.</span>, <span class="smallcaps">T.H. Parker</span>, <span class="smallcaps">S. Nakagawa</span>, <span class="smallcaps">J. Gurevitch</span>, <span class="smallcaps">Ecology</span>, and <span class="smallcaps">T.(Tools. for T. in Evolution) Working Group</span>. 2016. Promoting transparency in evolutionary biology and ecology. <em>Systematic Botany</em> 41: 495–497. Available at: <a href="http://www.bioone.org/doi/abs/10.1600/036364416X692262">http://www.bioone.org/doi/abs/10.1600/036364416X692262</a>.
</div>
<div id="ref-Soltis2023" class="csl-entry">
<span class="smallcaps">Soltis, D.E.</span>, <span class="smallcaps">V.B. Smocovitis</span>, <span class="smallcaps">K.K. Pham</span>, <span class="smallcaps">M.B.S. Cortez</span>, <span class="smallcaps">A.L. Smith</span>, and <span class="smallcaps">P.S. Soltis</span>. 2023. Rethinking the ph.d. Dissertation in botany: Widening the circle. <em>American Journal of Botany</em> 110: e16136. Available at: <a href="https://bsapubs.onlinelibrary.wiley.com/doi/abs/10.1002/ajb2.16136">https://bsapubs.onlinelibrary.wiley.com/doi/abs/10.1002/ajb2.16136</a>.
</div>
<div id="ref-Trisovic2022" class="csl-entry">
<span class="smallcaps">Trisovic, A.</span>, <span class="smallcaps">M.K. Lau</span>, <span class="smallcaps">T. Pasquier</span>, and <span class="smallcaps">M. Crosas</span>. 2022. A large-scale study on research code quality and execution. <em>Scientific Data</em> 9: 60. Available at: <a href="https://doi.org/10.1038/s41597-022-01143-6">https://doi.org/10.1038/s41597-022-01143-6</a>.
</div>
<div id="ref-Wagner2022" class="csl-entry">
<span class="smallcaps">Wagner, A.S.</span>, <span class="smallcaps">L.K. Waite</span>, <span class="smallcaps">M. Wierzba</span>, <span class="smallcaps">F. Hoffstaedter</span>, <span class="smallcaps">A.Q. Waite</span>, <span class="smallcaps">B. Poldrack</span>, <span class="smallcaps">S.B. Eickhoff</span>, and <span class="smallcaps">M. Hanke</span>. 2022. FAIRly big: A framework for computationally reproducible processing of large-scale data. <em>Scientific Data</em> 9: 80. Available at: <a href="https://doi.org/10.1038/s41597-022-01163-2">https://doi.org/10.1038/s41597-022-01163-2</a>.
</div>
<div id="ref-Wickham2017" class="csl-entry">
<span class="smallcaps">Wickham, H.</span>, and <span class="smallcaps">G. Grolemund</span>. 2017. R for data science: Import, tidy, transform, visualize, and model data. 1st ed. O’Reilly Media, Inc. Available at: <a href="http://r4ds.had.co.nz">http://r4ds.had.co.nz</a>.
</div>
<div id="ref-Williams2023" class="csl-entry">
<span class="smallcaps">Williams, J.W.</span>, <span class="smallcaps">A. Taylor</span>, <span class="smallcaps">K.A. Tolley</span>, <span class="smallcaps">D.B. Provete</span>, <span class="smallcaps">R. Correia</span>, <span class="smallcaps">T.B. Guedes</span>, <span class="smallcaps">H. Farooq</span>, et al. 2023. Shifts to open access with high article processing charges hinder research equity and careers. <em>Journal of Biogeography</em> 50: 1485–1489. Available at: <a href="https://doi.org/10.1111/jbi.14697">https://doi.org/10.1111/jbi.14697</a>.
</div>
<div id="ref-Wojahn2021" class="csl-entry">
<span class="smallcaps">Wojahn, J.M.A.</span>, <span class="smallcaps">S.J. Galla</span>, <span class="smallcaps">A.E. Melton</span>, and <span class="smallcaps">S. Buerki</span>. 2021. G2PMineR: A genome to phenome literature review approach. <em>Genes</em> 12: 293. Available at: <a href="https://doi.org/10.3390/genes12020293">https://doi.org/10.3390/genes12020293</a>.
</div>
</div>
</div>
<div id="appendix-1" class="section level1 appendix" number="20">
<h1><span class="header-section-number">20</span> Appendix 1</h1>
<p>Citations of all R packages used to generate this report.</p>
<p>[1] J. Allaire, Y. Xie, C. Dervieux, et al. <em>rmarkdown: Dynamic
Documents for R</em>. R package version 2.21. 2023.
<a href="https://CRAN.R-project.org/package=rmarkdown" class="uri">https://CRAN.R-project.org/package=rmarkdown</a>.</p>
<p>[2] C. Boettiger. <em>knitcitations: Citations for Knitr Markdown Files</em>.
R package version 1.0.12. 2021.
<a href="https://github.com/cboettig/knitcitations" class="uri">https://github.com/cboettig/knitcitations</a>.</p>
<p>[3] M. C. Koohafkan. <em>kfigr: Integrated Code Chunk Anchoring and
Referencing for R Markdown Documents</em>. R package version 1.2.1. 2021.
<a href="https://github.com/mkoohafkan/kfigr" class="uri">https://github.com/mkoohafkan/kfigr</a>.</p>
<p>[4] R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2022. <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>.</p>
<p>[5] H. Wickham, J. Bryan, and M. Barrett. <em>usethis: Automate Package
and Project Setup</em>. R package version 2.1.6. 2022.
<a href="https://CRAN.R-project.org/package=usethis" class="uri">https://CRAN.R-project.org/package=usethis</a>.</p>
<p>[6] H. Wickham, R. François, L. Henry, et al. <em>dplyr: A Grammar of Data
Manipulation</em>. R package version 1.0.9. 2022.
<a href="https://CRAN.R-project.org/package=dplyr" class="uri">https://CRAN.R-project.org/package=dplyr</a>.</p>
<p>[7] H. Wickham, J. Hester, W. Chang, et al. <em>devtools: Tools to Make
Developing R Packages Easier</em>. R package version 2.4.4. 2022.
<a href="https://CRAN.R-project.org/package=devtools" class="uri">https://CRAN.R-project.org/package=devtools</a>.</p>
<p>[8] Y. Xie. <em>bookdown: Authoring Books and Technical Documents with R
Markdown</em>. ISBN 978-1138700109. Boca Raton, Florida: Chapman and
Hall/CRC, 2016. <a href="https://bookdown.org/yihui/bookdown" class="uri">https://bookdown.org/yihui/bookdown</a>.</p>
<p>[9] Y. Xie. <em>bookdown: Authoring Books and Technical Documents with R
Markdown</em>. R package version 0.33. 2023.
<a href="https://CRAN.R-project.org/package=bookdown" class="uri">https://CRAN.R-project.org/package=bookdown</a>.</p>
<p>[10] Y. Xie. <em>Dynamic Documents with R and knitr</em>. 2nd. ISBN
978-1498716963. Boca Raton, Florida: Chapman and Hall/CRC, 2015.
<a href="https://yihui.org/knitr/" class="uri">https://yihui.org/knitr/</a>.</p>
<p>[11] Y. Xie. <em>formatR: Format R Code Automatically</em>. R package version
1.12. 2022. <a href="https://github.com/yihui/formatR" class="uri">https://github.com/yihui/formatR</a>.</p>
<p>[12] Y. Xie. “knitr: A Comprehensive Tool for Reproducible Research in
R”. In: <em>Implementing Reproducible Computational Research</em>. Ed. by V.
Stodden, F. Leisch and R. D. Peng. ISBN 978-1466561595. Chapman and
Hall/CRC, 2014.</p>
<p>[13] Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.42. 2023.
<a href="https://yihui.org/knitr/" class="uri">https://yihui.org/knitr/</a>.</p>
<p>[14] Y. Xie and J. Allaire. <em>tufte: Tufte’s Styles for R Markdown
Documents</em>. R package version 0.12. 2022.
<a href="https://github.com/rstudio/tufte" class="uri">https://github.com/rstudio/tufte</a>.</p>
<p>[15] Y. Xie, J. Allaire, and G. Grolemund. <em>R Markdown: The Definitive
Guide</em>. Boca Raton, Florida: Chapman and Hall/CRC, 2018. ISBN:
9781138359338. <a href="https://bookdown.org/yihui/rmarkdown" class="uri">https://bookdown.org/yihui/rmarkdown</a>.</p>
<p>[16] Y. Xie, C. Dervieux, and E. Riederer. <em>R Markdown Cookbook</em>. Boca
Raton, Florida: Chapman and Hall/CRC, 2020. ISBN: 9780367563837.
<a href="https://bookdown.org/yihui/rmarkdown-cookbook" class="uri">https://bookdown.org/yihui/rmarkdown-cookbook</a>.</p>
<p>[17] H. Zhu. <em>kableExtra: Construct Complex Table with kable and Pipe
Syntax</em>. R package version 1.3.4. 2021.
<a href="https://CRAN.R-project.org/package=kableExtra" class="uri">https://CRAN.R-project.org/package=kableExtra</a>.</p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
