<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Sven Buerki - Boise State University" />

<meta name="date" content="2023-08-21" />

<title>EEB 603 – Reproducible Science</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Reproducible Science</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Timetable.html">Timetable</a>
</li>
<li>
  <a href="Chapters.html">Chapters</a>
</li>
<li>
  <a href="Resources.html">Resources</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">EEB 603 – Reproducible Science</h1>
<h3 class="subtitle">Resources</h3>
<h4 class="author">Sven Buerki - Boise State University</h4>
<h4 class="date">2023-08-21</h4>

</div>


<script type="text/javascript">
$(document).ready(function() {
  $('#header').parent().prepend('<div id=\"logo\"><img src=\"Images/boisestate-primarylogo-2color-rgb.png\" style=\"position:absolute; top:0; right:0; padding:10px; height:120px\"></div>');
  $('#header').css('margin-right', '120px')
});
</script>
<div style="text-align: right">
<a href="Resources.pdf">Download pdf version</a>
</div>
<div id="introduction" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Introduction</h1>
<p>The field of reproducible science as implemented in <a href="https://rstudio.com/">RStudio</a> is moving very fast and it is hard to keep up with new R packages and tools. The instructor is attempting at providing students with relevant online resources to complement class materials.</p>
</div>
<div id="podcast" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Podcast</h1>
<p>The challenge of reproducing results from ten-year-old code:</p>
<ul>
<li><a href="https://www.nature.com/articles/d41586-020-02484-1" class="uri">https://www.nature.com/articles/d41586-020-02484-1</a></li>
</ul>
</div>
<div id="library-of-open-access-books" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Library of open-access books</h1>
<p>Please find <a href="https://bookdown.org">here</a> a suite of open-access books written with the R <a href="https://github.com/rstudio/bookdown"><em>bookdown</em> package</a>, which are relevant to the EEB603 - Reproducible Science course.</p>
</div>
<div id="r-markdown" class="section level1" number="4">
<h1><span class="header-section-number">4</span> R Markdown</h1>
<p>Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <a href="http://rmarkdown.rstudio.com" class="uri">http://rmarkdown.rstudio.com</a>. Please find below additional key resources on this topic:</p>
<ul>
<li><strong>R Markdown: The Definitive Guide:</strong> <a href="https://bookdown.org/yihui/rmarkdown/" class="uri">https://bookdown.org/yihui/rmarkdown/</a></li>
</ul>
</div>
<div id="writing-books-and-complexlarge-documents" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Writing books and complex/large documents</h1>
<p>R Markdown can be used to produce basic HTML, PDF, and Word documents; however, more complex and larger projects can become difficult to manage in a single R Markdown file. The <em>bookdown</em> package addresses this limitation, and offers several key improvements:</p>
<ul>
<li>Books and reports can be built from multiple R Markdown files.</li>
<li>Additional formatting features are added, such as cross-referencing, and numbering of figures, equations, and tables.</li>
<li>Documents can easily be exported in a range of formats suitable for publishing, including PDF, e-books and HTML websites.</li>
</ul>
<p>A book dedicated to this topic can be accessed <a href="https://bookdown.org/yihui/bookdown/">here</a> and the package can be downloaded <a href="https://CRAN.R-project.org/package=bookdown">here</a>.</p>
</div>
<div id="notebook" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Notebook</h1>
<p>An R Notebook is an R Markdown document with chunks that can be executed independently and interactively, with output visible immediately beneath the input.</p>
<p>You can create a new notebook in RStudio with the menu command <code>File -&gt; New File -&gt; R Notebook</code>, or by using the <code>html_notebook</code> output type in your document’s YAML metadata.</p>
</div>
<div id="homework-assignments-on-rpubs" class="section level1" number="7">
<h1><span class="header-section-number">7</span> Homework assignments on RPubs</h1>
<p>A huge number of homework assignments have been published to the website <a href="https://RPubs.com" class="uri">https://RPubs.com</a> (a free publishing platform provided by RStudio), which shows that R Markdown is easy and convenient enough for students to do their homework assignments. As a TA or GA, you might even consider asking your students to complete their homework using R Markdown and publishing them on RPubs!</p>
</div>
<div id="journals" class="section level1" number="8">
<h1><span class="header-section-number">8</span> Journals</h1>
<p>Submitting scientific manuscripts written in R Markdown is still challenging; however the R <a href="https://cran.r-project.org/web/packages/rticles/index.html"><em>rticles</em> package</a> was designed to simplify the creation of documents that conform to submission standards for academic journals. The package provides a suite of custom R Markdown LaTeX formats and templates for the following journals/publishers that are relevant to the EEB program:</p>
<ul>
<li>Biometrics articles</li>
<li>Elsevier journal submissions</li>
<li>Frontiers articles</li>
<li>MDPI journal submissions</li>
<li>PeerJ articles</li>
<li>PNAS articles</li>
<li>Royal Society Open Science journal submissions</li>
<li>Sage journal submissions</li>
<li>Springer journal submissions</li>
<li>The R Journal articles</li>
<li>Taylor &amp; Francis articles</li>
</ul>
</div>
<div id="preparing-literature" class="section level1" number="9">
<h1><span class="header-section-number">9</span> Preparing literature</h1>
<p>The <a href="https://www.doi2bib.org/">doi2bib</a> web tool allows you to generate BibTex formatted literature from DOIs.</p>
</div>
<div id="interactive-tutorials" class="section level1" number="10">
<h1><span class="header-section-number">10</span> Interactive tutorials</h1>
<p>The R <a href="https://cran.r-project.org/web/packages/learnr/index.html"><em>learnr</em> package</a> allows to turn R Markdown documents into interactive tutorials. Tutorials consist of content along with interactive components for checking and reinforcing understanding. Tutorials can include any or all of the following:</p>
<ul>
<li>Narrative, figures, illustrations, and equations.</li>
<li>Code exercises (R code chunks that users can edit and execute directly).</li>
<li>Quiz questions.</li>
<li>Videos (currently supported services include YouTube and Vimeo).</li>
<li>Interactive Shiny components.</li>
</ul>
<p>Please see the full documentation (with examples) at this url: <a href="https://rstudio.github.io/learnr/" class="uri">https://rstudio.github.io/learnr/</a></p>
</div>
<div id="fair-and-care-principles" class="section level1" number="11">
<h1><span class="header-section-number">11</span> FAIR and CARE Principles</h1>
<ul>
<li>ResearchHub: <a href="https://research-hub.auckland.ac.nz/guide-to-managing-research-data/ethics-integrity-and-compliance/the-fair-and-care-principles-for-research-data" class="uri">https://research-hub.auckland.ac.nz/guide-to-managing-research-data/ethics-integrity-and-compliance/the-fair-and-care-principles-for-research-data</a></li>
</ul>
</div>
<div id="special-issue-on-reproducibility-and-research-integrity" class="section level1" number="12">
<h1><span class="header-section-number">12</span> Special Issue on Reproducibility and Research Integrity</h1>
<p><em>BMC Research Notes</em> published a special issue on Reproducibility and research integrity in 2022. You can consult all the publications <a href="https://www.biomedcentral.com/collections/reproducibility">here</a>.</p>
</div>
<div id="publications-used-in-this-class" class="section level1" number="13">
<h1><span class="header-section-number">13</span> Publications used in this class</h1>
<p>Here is a list of the publications used in this class.</p>
<p>[1] M. Cargill and P. O’Connor. _ Writing Scientific Research
Articles: Strategy and Steps _. Wiley-Blackwell , .</p>
<p>[2] L. Allen, J. Scott, A. Brand, et al. “Publishing: Credit where
credit is due”. In: <em>Nature</em> 508 (2014), pp. 312-313. DOI:
10.1038/508312a.</p>
<p>[3] M. Baker. “1,500 scientists lift the lid on reproducibility”. In:
<em>Nature</em> 533.7604 (May. 2016), pp. 452-454. DOI: 10.1038/533452a.
<a href="https://doi.org/10.1038/533452a" class="uri">https://doi.org/10.1038/533452a</a>.</p>
<p>[4] R. Barron, P. Martinez, M. Serpe, et al. “Development of an In
Vitro Method of Propagation for Artemisia tridentata subsp. tridentata
to Support Genome Sequencing and Genotype-by-Environment Research”. In:
<em>Plants</em> 9.12 (2020). ISSN: 2223-7747. DOI: 10.3390/plants9121717.
<a href="https://www.mdpi.com/2223-7747/9/12/1717" class="uri">https://www.mdpi.com/2223-7747/9/12/1717</a>.</p>
<p>[5] R. E. Bone, J. A. C. Smith, N. Arrigo, et al. “A macro-ecological
perspective on crassulacean acid metabolism (CAM) photosynthesis
evolution in Afro-Madagascan drylands: Eulophiinae orchids as a case
study”. In: <em>New Phytologist</em> 208.2 (2015). 2015-19192, pp. 469-481.
ISSN: 1469-8137. DOI: 10.1111/nph.13572.
<a href="http://dx.doi.org/10.1111/nph.13572" class="uri">http://dx.doi.org/10.1111/nph.13572</a>.</p>
<p>[6] J. Boulter, M. L. Orozco Morales, N. Principe, et al. “What is
Kindness in Science and why does it matter?” In: <em>Immunology &amp; Cell
Biology</em> 101.2 (2023), pp. 97-103. DOI:
<a href="https://doi.org/10.1111/imcb.12580" class="uri">https://doi.org/10.1111/imcb.12580</a>. eprint:
<a href="https://onlinelibrary.wiley.com/doi/pdf/10.1111/imcb.12580" class="uri">https://onlinelibrary.wiley.com/doi/pdf/10.1111/imcb.12580</a>.
<a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/imcb.12580" class="uri">https://onlinelibrary.wiley.com/doi/abs/10.1111/imcb.12580</a>.</p>
<p>[7] British Ecological Society, ed. <em>A Guide to Data Management in
Ecology and Evolution</em>. British Ecological Society. 2014.</p>
<p>[8] British Ecological Society, ed. <em>A Guide to Getting Published in
Ecology and Evolution</em>. British Ecological Society. 2014.</p>
<p>[9] British Ecological Society, ed. <em>A Guide to Peer Review in Ecology
and Evolution</em>. British Ecological Society. 2014.</p>
<p>[10] British Ecological Society, ed. <em>A Guide to Reproducible Code in
Ecology and Evolution</em>. British Ecological Society. 2014.</p>
<p>[11] S. R. Carroll, E. Herczog, M. Hudson, et al. “Operationalizing the
CARE and FAIR Principles for Indigenous data futures”. In: <em>Scientific
Data</em> 8.1 (Apr. 2021), p. 108. ISSN: 2052-4463. DOI:
10.1038/s41597-021-00892-0.
<a href="https://doi.org/10.1038/s41597-021-00892-0" class="uri">https://doi.org/10.1038/s41597-021-00892-0</a>.</p>
<p>[12] H. J. Cole, D. G. Gomes, and J. R. Barber. “EcoCountHelper: an R
package and analytical pipeline for the analysis of ecological count
data using GLMMs, and a case study of bats in Grant Teton National
Park”. In: <em>PeerJ</em> 10 (Dec. 2022), p. e14509. DOI: 10.7717/peerj.14509.
<a href="https://doi.org/10.7717/peerj.14509" class="uri">https://doi.org/10.7717/peerj.14509</a>.</p>
<p>[13] D. L. Donoho. “An invitation to reproducible computational
research”. In: <em>Biostatistics</em> 11.3 (2010), pp. 385-388. DOI:
10.1093/biostatistics/kxq028. eprint:
/oup/backfile/content_public/journal/biostatistics/11/3/10.1093/biostatistics/kxq028/2/kxq028.pdf.
<a href="http://dx.doi.org/10.1093/biostatistics/kxq028" class="uri">http://dx.doi.org/10.1093/biostatistics/kxq028</a>.</p>
<p>[14] P. Ellestad, M. A. Pérez-Farrera, and S. Buerki. “Genomic Insights
into Cultivated Mexican Vanilla planifolia Reveal High Levels of
Heterozygosity Stemming from Hybridization”. In: <em>Plants</em> 11.16 (Aug.
2022), p. 2090. DOI: 10.3390/plants11162090.
<a href="https://doi.org/10.3390/plants11162090" class="uri">https://doi.org/10.3390/plants11162090</a>.</p>
<p>[15] C. W. Fox and C. S. Burns. “The relationship between manuscript
title structure and success: editorial decisions and citation
performance for an ecological journal”. In: <em>Ecology and Evolution</em>
5.10 (2015), pp. 1970-1980. DOI: 10.1002/ece3.1480. eprint:
<a href="https://onlinelibrary.wiley.com/doi/pdf/10.1002/ece3.1480" class="uri">https://onlinelibrary.wiley.com/doi/pdf/10.1002/ece3.1480</a>.
<a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/ece3.1480" class="uri">https://onlinelibrary.wiley.com/doi/abs/10.1002/ece3.1480</a>.</p>
<p>[16] L. P. Freedman, I. M. Cockburn, and T. S. Simcoe. “The Economics
of Reproducibility in Preclinical Research”. In: <em>PLOS Biology</em> 13.6
(Jun. 2015), p. e1002165. DOI: 10.1371/journal.pbio.1002165.
<a href="https://doi.org/10.1371/journal.pbio.1002165" class="uri">https://doi.org/10.1371/journal.pbio.1002165</a>.</p>
<p>[17] C. Gandrud. <em>Reproducible Research with R and RStudio</em>. Ed. by C.
Gandrud. CRC Press, 2015, p. 294. ISBN: 1466572841.</p>
<p>[18] C. Gandrud. <em>repmis: Miscellaneous Tools for Reproducible
Research</em>. R package version 0.5. 2016.
<a href="https://CRAN.R-project.org/package=repmis" class="uri">https://CRAN.R-project.org/package=repmis</a>.</p>
<p>[19] Q. Groom, L. Weatherdon, and I. R. Geijzendorffer. “Is citizen
science an open science in the case of biodiversity observations?” In:
<em>Journal of Applied Ecology</em> 54.2 (2017), pp. 612-617. DOI:
<a href="https://doi.org/10.1111/1365-2664.12767" class="uri">https://doi.org/10.1111/1365-2664.12767</a>. eprint:
<a href="https://besjournals.onlinelibrary.wiley.com/doi/pdf/10.1111/1365-2664.12767" class="uri">https://besjournals.onlinelibrary.wiley.com/doi/pdf/10.1111/1365-2664.12767</a>.
<a href="https://besjournals.onlinelibrary.wiley.com/doi/abs/10.1111/1365-2664.12767" class="uri">https://besjournals.onlinelibrary.wiley.com/doi/abs/10.1111/1365-2664.12767</a>.</p>
<p>[20] Y. Guangchuang, S. D. K., Z. Huachen, et al. “ggtree: an r package
for visualization and annotation of phylogenetic trees with their
covariates and other associated data”. In: <em>Methods in Ecology and
Evolution</em> 8.1 (2017), pp. 28-36. DOI: 10.1111/2041-210X.12628. eprint:
<a href="https://besjournals.onlinelibrary.wiley.com/doi/pdf/10.1111/2041-210X.12628" class="uri">https://besjournals.onlinelibrary.wiley.com/doi/pdf/10.1111/2041-210X.12628</a>.
<a href="https://besjournals.onlinelibrary.wiley.com/doi/abs/10.1111/2041-210X.12628" class="uri">https://besjournals.onlinelibrary.wiley.com/doi/abs/10.1111/2041-210X.12628</a>.</p>
<p>[21] J. Hester, G. Csárdi, H. Wickham, et al. <em>remotes: R Package
Installation from Remote Repositories, Including GitHub</em>. R package
version 2.1.0. 2019. <a href="https://github.com/r-lib/remotes#readme" class="uri">https://github.com/r-lib/remotes#readme</a>.</p>
<p>[22] N. L. Kerr. “HARKing: Hypothesizing After the Results are Known”.
In: <em>Personality and Social Psychology Review</em> 2.3 (1998), pp. 196-217.
<a href="https://doi.org/10.1207/s15327957pspr0203_4" class="uri">https://doi.org/10.1207/s15327957pspr0203_4</a>.</p>
<p>[23] S. Y. Khoo. “Article Processing Charge Hyperinflation and Price
Insensitivity: An Open Access Sequel to the Serials Crisis”. In: <em>LIBER
Quarterly: The Journal of the Association of European Research
Libraries</em> 29.1 (May. 2019), p. 1–18. DOI: 10.18352/lq.10280.
<a href="https://liberquarterly.eu/article/view/10729" class="uri">https://liberquarterly.eu/article/view/10729</a>.</p>
<p>[24] F. Markowetz. “Five selfish reasons to work reproducibly”. In:
<em>Genome Biology</em> 16.1 (Dec. 2015), p. 274. ISSN: 1474-760X. DOI:
10.1186/s13059-015-0850-7. <a href="https://doi.org/10.1186/s13059-015-0850-7" class="uri">https://doi.org/10.1186/s13059-015-0850-7</a>.</p>
<p>[25] M. R. Munafo, B. A. Nosek, D. V. M. Bishop, et al. “A manifesto
for reproducible science”. English. In: <em>Nature human behaviour</em> 1.0021
(Jan. 2017), p. 0021. ISSN: 2397-3374. DOI: 10.1038/s41562-016-0021.</p>
<p>[26] B. A. Nosek, G. Alter, G. C. Banks, et al. “Promoting an open
research culture”. In: <em>Science</em> 348.6242 (2015), pp. 1422-1425. ISSN:
0036-8075. DOI: 10.1126/science.aab2374. eprint:
<a href="http://science.sciencemag.org/content/348/6242/1422.full.pdf" class="uri">http://science.sciencemag.org/content/348/6242/1422.full.pdf</a>.
<a href="http://science.sciencemag.org/content/348/6242/1422" class="uri">http://science.sciencemag.org/content/348/6242/1422</a>.</p>
<p>[27] B. A. Nosek, C. R. Ebersole, A. C. DeHaven, et al. “The
preregistration revolution”. In: <em>Proceedings of the National Academy
of Sciences</em> 115.11 (2018), pp. 2600-2606. ISSN: 0027-8424. DOI:
10.1073/pnas.1708274114. eprint:
<a href="http://www.pnas.org/content/115/11/2600.full.pdf" class="uri">http://www.pnas.org/content/115/11/2600.full.pdf</a>.
<a href="http://www.pnas.org/content/115/11/2600" class="uri">http://www.pnas.org/content/115/11/2600</a>.</p>
<p>[28] B. A. Nosek, J. R. Spies, and M. Motyl. “Scientific Utopia: II.
Restructuring Incentives and Practices to Promote Truth Over
Publishability”. In: <em>Perspectives on Psychological Science</em> 7.6
(2012), pp. 615-631. DOI: 10.1177/1745691612459058.
<a href="https://doi.org/10.1177/1745691612459058" class="uri">https://doi.org/10.1177/1745691612459058</a>.</p>
<p>[29] R. D. Peng and S. C. Hicks. “Reproducible Research: A
Retrospective”. In: <em>Annual Review of Public Health</em> 42.1 (2021). PMID:
33467923, pp. 79-93. DOI: 10.1146/annurev-publhealth-012420-105110.
eprint: <a href="https://doi.org/10.1146/annurev-publhealth-012420-105110" class="uri">https://doi.org/10.1146/annurev-publhealth-012420-105110</a></p>
<p>. <a href="https://doi.org/10.1146/annurev-publhealth-012420-105110" class="uri">https://doi.org/10.1146/annurev-publhealth-012420-105110</a>.</p>
<p>[30] R Core Team. <em>foreign: Read Data Stored by ‘Minitab’, ‘S’, ‘SAS’,
‘SPSS’, ‘Stata’, ‘Systat’, ‘Weka’, ‘dBase’, …</em> R package version
0.8-75. 2020. <a href="https://CRAN.R-project.org/package=foreign" class="uri">https://CRAN.R-project.org/package=foreign</a>.</p>
<p>[31] R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2019. <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>.</p>
<p>[32] D. Sarewitz. “The pressure to publish pushes down quality”. In:
<em>Nature</em> 533.7602 (May. 2016), pp. 147-147. DOI: 10.1038/533147a.
<a href="https://doi.org/10.1038/533147a" class="uri">https://doi.org/10.1038/533147a</a>.</p>
<p>[33] J. F. Smith, T. H. Parker, S. Nakagawa, et al. “Promoting
Transparency in Evolutionary Biology and Ecology”. In: <em>Systematic
Botany</em> 41.3 (Jul. 2016), pp. 495-497. ISSN: 0363-6445. DOI:
10.1600/036364416X692262.
<a href="http://www.bioone.org/doi/abs/10.1600/036364416X692262" class="uri">http://www.bioone.org/doi/abs/10.1600/036364416X692262</a>.</p>
<p>[34] D. E. Soltis, V. B. Smocovitis, K. K. Pham, et al. “Rethinking the
Ph.D. dissertation in botany: Widening the circle”. In: <em>American
Journal of Botany</em> 110.3 (2023), p. e16136. DOI:
<a href="https://doi.org/10.1002/ajb2.16136" class="uri">https://doi.org/10.1002/ajb2.16136</a>. eprint:
<a href="https://bsapubs.onlinelibrary.wiley.com/doi/pdf/10.1002/ajb2.16136" class="uri">https://bsapubs.onlinelibrary.wiley.com/doi/pdf/10.1002/ajb2.16136</a>.
<a href="https://bsapubs.onlinelibrary.wiley.com/doi/abs/10.1002/ajb2.16136" class="uri">https://bsapubs.onlinelibrary.wiley.com/doi/abs/10.1002/ajb2.16136</a>.</p>
<p>[35] A. Trisovic, M. K. Lau, T. Pasquier, et al. “A large-scale study
on research code quality and execution”. In: <em>Scientific Data</em> 9.1
(Feb. 2022), p. 60. ISSN: 2052-4463. DOI: 10.1038/s41597-022-01143-6.
<a href="https://doi.org/10.1038/s41597-022-01143-6" class="uri">https://doi.org/10.1038/s41597-022-01143-6</a>.</p>
<p>[36] J. Troudet, R. Vignes-Lebbe, P. Grandcolas, et al. “The Increasing
Disconnection of Primary Biodiversity Data from Specimens: How Does It
Happen and How to Handle It?” In: <em>Systematic Biology</em> (2018), p.
syy044. DOI: 10.1093/sysbio/syy044. eprint:
/oup/backfile/content_public/journal/sysbio/pap/10.1093_sysbio_syy044/3/syy044.pdf.
<a href="http://dx.doi.org/10.1093/sysbio/syy044" class="uri">http://dx.doi.org/10.1093/sysbio/syy044</a>.</p>
<p>[37] K. Wagenknecht, T. Woods, F. G. Sanz, et al. “EU-Citizen.Science:
A Platform for Mainstreaming Citizen Science and Open Science in
Europe”. In: <em>Data Intelligence</em> 3.1 (Feb. 2021), pp. 136-149. ISSN:
2641-435X. DOI: 10.1162/dint_a_00085. eprint:
<a href="https://direct.mit.edu/dint/article-pdf/3/1/136/1893818/dint\_a\_00085.pdf" class="uri">https://direct.mit.edu/dint/article-pdf/3/1/136/1893818/dint\_a\_00085.pdf</a>.
<a href="https://doi.org/10.1162/dint%5C_a%5C_00085" class="uri">https://doi.org/10.1162/dint%5C_a%5C_00085</a>.</p>
<p>[38] A. S. Wagner, L. K. Waite, M. Wierzba, et al. “FAIRly big: A
framework for computationally reproducible processing of large-scale
data”. In: <em>Scientific Data</em> 9.1 (Mar. 2022), p. 80. ISSN: 2052-4463.
DOI: 10.1038/s41597-022-01163-2.
<a href="https://doi.org/10.1038/s41597-022-01163-2" class="uri">https://doi.org/10.1038/s41597-022-01163-2</a>.</p>
<p>[39] H. Wickham. <em>Advanced R</em>. Chapman &amp; Hall/CRC The R Series. Taylor
&amp; Francis, 2014. ISBN: 9781466586963.
<a href="https://books.google.com/books?id=PFHFNAEACAAJ" class="uri">https://books.google.com/books?id=PFHFNAEACAAJ</a>.</p>
<p>[40] H. Wickham and J. Bryan. <em>readxl: Read Excel Files</em>. R package
version 1.3.1. 2019. <a href="https://CRAN.R-project.org/package=readxl" class="uri">https://CRAN.R-project.org/package=readxl</a>.</p>
<p>[41] H. Wickham and G. Grolemund. <em>R for Data Science: Import, Tidy,
Transform, Visualize, and Model Data</em>. 1st. O’Reilly Media, Inc., 2017.
ISBN: 1491910399, 9781491910399. <a href="http://r4ds.had.co.nz" class="uri">http://r4ds.had.co.nz</a>.</p>
<p>[42] M. D. Wilkinson, M. Dumontier, I. J. Aalbersberg, et al. “The FAIR
Guiding Principles for scientific data management and stewardship”. In:
<em>Scientific Data</em> 3.1 (Mar. 2016), p. 160018. ISSN: 2052-4463. DOI:
10.1038/sdata.2016.18. <a href="https://doi.org/10.1038/sdata.2016.18" class="uri">https://doi.org/10.1038/sdata.2016.18</a>.</p>
<p>[43] J. W. Williams, A. Taylor, K. A. Tolley, et al. “Shifts to open
access with high article processing charges hinder research equity and
careers”. In: <em>Journal of Biogeography</em> 50.9 (Jul. 2023), pp.
1485-1489. DOI: 10.1111/jbi.14697. <a href="https://doi.org/10.1111/jbi.14697" class="uri">https://doi.org/10.1111/jbi.14697</a>.</p>
<p>[44] J. M. A. Wojahn, S. J. Galla, A. E. Melton, et al. “G2PMineR: A
Genome to Phenome Literature Review Approach”. In: <em>Genes</em> 12.2 (Feb.
2021), p. 293. DOI: 10.3390/genes12020293.
<a href="https://doi.org/10.3390/genes12020293" class="uri">https://doi.org/10.3390/genes12020293</a>.</p>
<div id="refs">

</div>
</div>
<div id="appendix-1" class="section level1 appendix" number="14">
<h1><span class="header-section-number">14</span> Appendix 1</h1>
<p>Citations of all R packages used to generate this report.</p>
<p>[1] J. Allaire, Y. Xie, C. Dervieux, et al. <em>rmarkdown: Dynamic
Documents for R</em>. R package version 2.21. 2023.
<a href="https://CRAN.R-project.org/package=rmarkdown" class="uri">https://CRAN.R-project.org/package=rmarkdown</a>.</p>
<p>[2] C. Boettiger. <em>knitcitations: Citations for Knitr Markdown Files</em>.
R package version 1.0.12. 2021.
<a href="https://github.com/cboettig/knitcitations" class="uri">https://github.com/cboettig/knitcitations</a>.</p>
<p>[3] M. C. Koohafkan. <em>kfigr: Integrated Code Chunk Anchoring and
Referencing for R Markdown Documents</em>. R package version 1.2.1. 2021.
<a href="https://github.com/mkoohafkan/kfigr" class="uri">https://github.com/mkoohafkan/kfigr</a>.</p>
<p>[4] R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2022. <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>.</p>
<p>[5] H. Wickham, J. Bryan, and M. Barrett. <em>usethis: Automate Package
and Project Setup</em>. R package version 2.1.6. 2022.
<a href="https://CRAN.R-project.org/package=usethis" class="uri">https://CRAN.R-project.org/package=usethis</a>.</p>
<p>[6] H. Wickham, R. François, L. Henry, et al. <em>dplyr: A Grammar of Data
Manipulation</em>. R package version 1.0.9. 2022.
<a href="https://CRAN.R-project.org/package=dplyr" class="uri">https://CRAN.R-project.org/package=dplyr</a>.</p>
<p>[7] H. Wickham, J. Hester, W. Chang, et al. <em>devtools: Tools to Make
Developing R Packages Easier</em>. R package version 2.4.4. 2022.
<a href="https://CRAN.R-project.org/package=devtools" class="uri">https://CRAN.R-project.org/package=devtools</a>.</p>
<p>[8] Y. Xie. <em>bookdown: Authoring Books and Technical Documents with R
Markdown</em>. ISBN 978-1138700109. Boca Raton, Florida: Chapman and
Hall/CRC, 2016. <a href="https://bookdown.org/yihui/bookdown" class="uri">https://bookdown.org/yihui/bookdown</a>.</p>
<p>[9] Y. Xie. <em>bookdown: Authoring Books and Technical Documents with R
Markdown</em>. R package version 0.33. 2023.
<a href="https://CRAN.R-project.org/package=bookdown" class="uri">https://CRAN.R-project.org/package=bookdown</a>.</p>
<p>[10] Y. Xie. <em>Dynamic Documents with R and knitr</em>. 2nd. ISBN
978-1498716963. Boca Raton, Florida: Chapman and Hall/CRC, 2015.
<a href="https://yihui.org/knitr/" class="uri">https://yihui.org/knitr/</a>.</p>
<p>[11] Y. Xie. <em>formatR: Format R Code Automatically</em>. R package version
1.12. 2022. <a href="https://github.com/yihui/formatR" class="uri">https://github.com/yihui/formatR</a>.</p>
<p>[12] Y. Xie. “knitr: A Comprehensive Tool for Reproducible Research in
R”. In: <em>Implementing Reproducible Computational Research</em>. Ed. by V.
Stodden, F. Leisch and R. D. Peng. ISBN 978-1466561595. Chapman and
Hall/CRC, 2014.</p>
<p>[13] Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.42. 2023.
<a href="https://yihui.org/knitr/" class="uri">https://yihui.org/knitr/</a>.</p>
<p>[14] Y. Xie and J. Allaire. <em>tufte: Tufte’s Styles for R Markdown
Documents</em>. R package version 0.12. 2022.
<a href="https://github.com/rstudio/tufte" class="uri">https://github.com/rstudio/tufte</a>.</p>
<p>[15] Y. Xie, J. Allaire, and G. Grolemund. <em>R Markdown: The Definitive
Guide</em>. Boca Raton, Florida: Chapman and Hall/CRC, 2018. ISBN:
9781138359338. <a href="https://bookdown.org/yihui/rmarkdown" class="uri">https://bookdown.org/yihui/rmarkdown</a>.</p>
<p>[16] Y. Xie, C. Dervieux, and E. Riederer. <em>R Markdown Cookbook</em>. Boca
Raton, Florida: Chapman and Hall/CRC, 2020. ISBN: 9780367563837.
<a href="https://bookdown.org/yihui/rmarkdown-cookbook" class="uri">https://bookdown.org/yihui/rmarkdown-cookbook</a>.</p>
<p>[17] H. Zhu. <em>kableExtra: Construct Complex Table with kable and Pipe
Syntax</em>. R package version 1.3.4. 2021.
<a href="https://CRAN.R-project.org/package=kableExtra" class="uri">https://CRAN.R-project.org/package=kableExtra</a>.</p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
